Makefile

```diff
@@ -1,30 +1,41 @@
 install:
+	@export PR=https://api.github.com/repos/$$TRAVIS_REPO_SLUG/pulls/$$TRAVIS_PULL_REQUEST; \
+	export BRANCH=$$(if [ "$$TRAVIS_PULL_REQUEST" == "false" ]; then echo $$TRAVIS_BRANCH; else echo `curl -s $$PR | jq -r .head.ref`; fi); \
+	echo "TRAVIS_BRANCH=$$TRAVIS_BRANCH, PR=$$PR, BRANCH=$$BRANCH"; \
+	pattern="refactor_"; \
+	if [[ $$BRANCH =~ $$pattern ]]; then \
+		repo=github.com/dedis/cosi; \
+		go get $$repo; \
+		cd $$GOPATH/src/$$repo; \
+		git checkout -f $BRANCH; \
+	fi;\
+	cd $$GOPATH/src/github.com/dedis/cothority; \
 	go get -t ./...
 
 test_fmt:
 	@echo Checking correct formatting of files
 	@{ \
-	files=$$( go fmt ./... ); \
-	if [ -n "$$files" ]; then \
+		files=$$( go fmt ./... ); \
```

Comment:
I think it even makes the test not working - `exit 1` will most probably just exit the sub-process generated by `{}`. This is why the linter didn't complain...